facebook/OpenBIC
# yv4: sd: Support DIMMs' temperature

# Description:
	BIC can monitor DIMMs' temperature and BMC can read them.

# Motivation:
	Support to monitor DIMMs' temperature.

# Test plan:
	- Use busctl to check DIMMs' information: Pass
	- DIMM mux will be switched to CPU when CPU reset : Pass
	- DIMM mux will be switched CPU after DC power-on and before post complete: Pass - Check DIMMs' presence and disable monitoring for unplugged ones: Pass

Test log:
- Use busctl to check DIMMs' information: root@bmc:~# busctl tree xyz.openbmc_project.PLDM `- /xyz `- /xyz/openbmc_project ``` | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_A_TEMP_C_5_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_B_TEMP_C_6_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_C_TEMP_C_7_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_D_TEMP_C_8_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_E_TEMP_C_9_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_F_TEMP_C_10_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_G_TEMP_C_11_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_H_TEMP_C_12_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_I_TEMP_C_13_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_J_TEMP_C_14_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_K_TEMP_C_15_134 | |- /xyz/openbmc_project/sensors/temperature/MB_DIMM_L_TEMP_C_16_134 ``` root@bmc:~# busctl introspect xyz.openbmc_project.PLDM /xyz/openbmc_project/sensors/temperature/MB_DIMM_A_TEMP_C_5_134
	NAME                                                  TYPE      SIGNATURE RESULT/VALUE                             FLAGS
	org.freedesktop.DBus.Introspectable                   interface -         -                                        -
	.Introspect                                           method    -         s                                        -
	org.freedesktop.DBus.Peer                             interface -         -                                        -
	```
	.Value                                                property  d         45                                       emits-change writable
	```

- DIMM mux will be switched to CPU when CPU reset: /* CPU reset*/ [00:12:41.776,000] <wrn> power_status: POST_COMPLETE: no uart:~$ i2c read I2C_4 0x21 0x0c 1 00000000: 00 /* DIMM mux is switched to CPU */

- DIMM mux will be switched to CPU when BIOS trains DIMM: /*BIOS is training DIMMs when DC on*/ [00:06:46.112,000] <wrn> power_status: DC_STATUS: on uart:~$ i2c read I2C_4 0x21 0x0c 1 00000000: 00 /* DIMM mux is switched to CPU */

- Check DIMMs' presence and disable monitoring for unplugged ones: /* Only Install DIMMA */ user@ber300-f5a-dhcp:~$  curl -u root:0penBmc -k https://192.168.88.18/redfish/v1/Chassis/PLDM_Device_134/Sensors/temperature_MB_DIMM_A_TEMP_C_5_134 { "@odata.id": "/redfish/v1/Chassis/PLDM_Device_134/Sensors/temperature_MB_DIMM_A_TEMP_C_5_134", "@odata.type": "#Sensor.v1_2_0.Sensor", "Id": "temperature_MB_DIMM_A_TEMP_C_5_134", "Name": "MB DIMM A TEMP C 5 134", "Reading": 36.0, "ReadingRangeMax": 255.0, "ReadingRangeMin": 0.0, ``` /*the Other DIMM*/ user@ber300-f5a-dhcp:~$  curl -u root:0penBmc -k https://192.168.88.18/redfish/v1/Chassis/PLDM_Device_134/Sensors/temperature_MB_DIMM_B_TEMP_C_6_134 { "@odata.id": "/redfish/v1/Chassis/PLDM_Device_134/Sensors/temperature_MB_DIMM_B_TEMP_C_6_134", "@odata.type": "#Sensor.v1_2_0.Sensor", "Id": "temperature_MB_DIMM_B_TEMP_C_6_134", "Name": "MB DIMM B TEMP C 6 134", "Reading": null, "ReadingRangeMax": 255.0, "ReadingRangeMin": 0.0, ```
